src/main/scala/search/Search.scala

```diff
@@ -0,0 +1,10 @@
+package basic
+package search
+
+import fs2.Stream
+
+trait Search[F[_]] {
+
+  def search(c: Constraints): Stream[F, ObservingMode] // something like this?
```

Comment:
What do we get from putting these in a `Stream` as opposed to something like just  `F[List[ObservingMode]]`?  Even if the constraints somehow matched all available modes we're talking maybe generously 100s of modes not even 1000s?

Reply:
That sounds right. if we don't expect so many nodes. Could this eventually return ALL observations ever?

Reply:
No, it's just returning "modes" which are at least somewhat analogous to templates or blueprints in the `ocs` codebase.

Reply:
Yeah it may not make sense for it to be a stream, I don't know yet.